File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int[] answer = n % 2 == 0 ? new int[n/2] : new int[n/2+1]; 
        int tmp = 0;
        
        for(int i=0;i<=n;i++){
            if(i % 2 == 1){
                answer[tmp] = i;
                tmp++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    int* answer = (int*)malloc(((n/2) + 1) * sizeof(int));
    int len = 0;
    
    for(int i=1;i<=n;i+=2){
            answer[len] = i;
            len++;
        }
    return answer;
}